jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -625,10 +626,10 @@ private void addJacksonModule(SchemaGeneratorConfigBuilder configBuilder, Genera
      * @param file The file to write to
      * @throws MojoExecutionException In case of problems when writing the targeted file
      */
-    private void writeToFile(JsonNode jsonSchema, File file) throws MojoExecutionException {
+    private void writeToFile(ObjectMapper mapper, JsonNode jsonSchema, File file) throws MojoExecutionException {
         try (FileOutputStream outputStream = new FileOutputStream(file);
                 PrintWriter writer = new PrintWriter(new OutputStreamWriter(outputStream, StandardCharsets.UTF_8))) {
-            writer.print(jsonSchema.toPrettyString());
+            writer.print(mapper.writeValueAsString(jsonSchema));
```

Comment:
**question**: do we need to use the `.writerWithDefaultPrettyPrinter()` in order to still format/indent the output as before? I suppose for YAML, that's irrelevant – or does it then again return JSON?
```suggestion
            writer.print(mapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonSchema));
```

Reply:
Turns out the repo has good unit tests lol - it was failing on my changes because the default ObjectMapper does not pretty-print the json. I did some research, and rather than hardcoding `writerWithDefaultPrettyPrinter()` in the print method, we can toggle the pretty-print option on the ObjectMapper directly:
```
ObjectMapper mapper = new ObjectMapper()
                .enable(SerializationFeature.INDENT_OUTPUT);
```

Now, if someone uses your library and they *don't* want pretty-printing, all they have to do is simply provide their own default ObjectMapper and the pretty-printing won't happen. Thus we knock out 2 problems with this change 🚀 

I've thus updated this PR with that change

Reply:
I was considering that too, but that'd be a breaking change for people using a custom `ObjectMapper` already (where no pretty printing is enabled), who still expect the pretty printed output.

Writing this down now, I realise: that would be mostly relevant for users of the Maven plugin, who weren't able to use a custom `ObjectMapper` until now. 😅

I guess it's fine then.